main/management/commands/donor_import.py

```diff
@@ -43,10 +43,10 @@ def add_arguments(self, parser):
 
     def decode_subject(self, subject):
         subject = decode_header(subject)
-        default_charset = 'ASCII'
+        default_charset = 'utf-8'
         # Convert the list of tuples containing the decoded string and encoding to
-        # the same encoding.
-        return u''.join([unicode(s[0], s[1] or default_charset) for s in subject])
+        # UTF-8
+        return u''.join([s[0].encode(default_charset, 'replace').decode(default_charset, 'replace') for s in subject])
```

Comment:
This doesn't seem to be correct, tested what it should be:

```
-        return u''.join([s[0].encode(default_charset, 'replace').decode(default_charset, 'replace') for s in subject])
+        return ''.join([codecs.decode(s[0], s[1] or default_charset) for s in subject])
```

Requires an codecs import.

The test should be adjusted, main/tests/test_donor_import.py:
```
-        subject = str(Header(text, 'utf-8'))
+        subject = Header(text, 'utf-8')
```

Since s[0] are bytes.

Reply:
Agreed. I'll make your suggested changes and test. This one is tricky to test, so I'll have to rely on the unit tests.

Reply:
I've tested my change with our data set and it works

Reply:
OK - I pushed another commit with your fix.